src/widget/wtracktext.cpp

```diff
@@ -1,109 +1,20 @@
-        WTrackMenu::Feature::Playlist |
-        WTrackMenu::Feature::Crate |
-        WTrackMenu::Feature::Metadata |
-        WTrackMenu::Feature::Reset |
-        WTrackMenu::Feature::Analyze |
-        WTrackMenu::Feature::BPM |
-        WTrackMenu::Feature::Color |
-        WTrackMenu::Feature::FileBrowser |
-        WTrackMenu::Feature::Properties |
-        WTrackMenu::Feature::UpdateReplayGainFromPregain |
-        WTrackMenu::Feature::FindOnWeb |
-        WTrackMenu::Feature::SelectInLibrary;
-} // namespace
-
 WTrackText::WTrackText(QWidget* pParent,
         UserSettingsPointer pConfig,
         Library* pLibrary,
         const QString& group)
-        : WLabel(pParent),
-          m_group(group),
-          m_pConfig(pConfig),
-          m_pTrackMenu(make_parented<WTrackMenu>(
-                  this, pConfig, pLibrary, kTrackMenuFeatures)) {
-    setAcceptDrops(true);
+        : WTrackProperty(pParent, pConfig, pLibrary, group) {
 }
 
 WTrackText::~WTrackText() {
     // Required to allow forward declaration of WTrackMenu in header
 }
 
-void WTrackText::slotTrackLoaded(TrackPointer pTrack) {
-    if (!pTrack) {
-        return;
-    }
-    m_pCurrentTrack = pTrack;
-    connect(pTrack.get(),
-            &Track::changed,
-            this,
-            &WTrackText::slotTrackChanged);
-    updateLabel();
-}
-
-void WTrackText::slotLoadingTrack(TrackPointer pNewTrack, TrackPointer pOldTrack) {
-    Q_UNUSED(pNewTrack);
-    Q_UNUSED(pOldTrack);
-    if (m_pCurrentTrack) {
-        disconnect(m_pCurrentTrack.get(), nullptr, this, nullptr);
-    }
-    m_pCurrentTrack.reset();
-    updateLabel();
-}
-
-void WTrackText::slotTrackChanged(TrackId trackId) {
-    Q_UNUSED(trackId);
-    updateLabel();
-}
-
-void WTrackText::updateLabel() {
-    if (m_pCurrentTrack) {
-        setText(m_pCurrentTrack->getInfo());
-    } else {
-        setText("");
-    }
-}
-
-void WTrackText::mouseMoveEvent(QMouseEvent *event) {
-    if (event->buttons().testFlag(Qt::LeftButton) && m_pCurrentTrack) {
-        DragAndDropHelper::dragTrack(m_pCurrentTrack, this, m_group);
-    }
-}
-
-void WTrackText::mouseDoubleClickEvent(QMouseEvent* event) {
-    Q_UNUSED(event);
-    if (m_pCurrentTrack) {
-        m_pTrackMenu->loadTrack(m_pCurrentTrack, m_group);
-        m_pTrackMenu->slotShowDlgTrackInfo();
-    }
-}
-
-void WTrackText::dragEnterEvent(QDragEnterEvent *event) {
-    DragAndDropHelper::handleTrackDragEnterEvent(event, m_group, m_pConfig);
-}
-
-void WTrackText::dropEvent(QDropEvent *event) {
-    DragAndDropHelper::handleTrackDropEvent(event, *this, m_group, m_pConfig);
-}
-
-void WTrackText::contextMenuEvent(QContextMenuEvent* event) {
-    event->accept();
-    if (m_pCurrentTrack) {
-        m_pTrackMenu->loadTrack(m_pCurrentTrack, m_group);
-        // Create the right-click menu
-        m_pTrackMenu->popup(event->globalPos());
-    }
+void WTrackText::setup(const QDomNode& node, const SkinContext& context) {
+    WLabel::setup(node, context);
```

Comment:
as rightfully pointed out by clazy.
```suggestion
    WTrackProperty::setup(node, context);
```

Reply:
nope, false positive: we do exactly **not** want that setup which parses the widget context for `Property` node since WTrackText is explicitly trackInfo not some specific tag.

Reply:
How can we let clazy know about it?

Reply:
See https://github.com/KDE/clazy/blob/master/README.md#reducing-warning-noise
Feel free to reformulate the comment. 
```suggestion
    // we do exactly not want that setup which parses the widget context 
    // for Property node since WTrackText is explicitly trackInfo not some specific tag.
    WLabel::setup(node, context); // clazy:exclude=skipped-base-method
```

Reply:
Thanks! Was on mobile so looking this up was rather cumbersome.
Fixed.

Reply:
Note that this decision results in the violation of the Liskov substitution principle. So that is an indicator that inheritance is probably not the best solution for achieving code-reuse here...

Reply:
So (theoretically) we would do it the other way around?
* base class is WTracktext, uses `trackInfo` and the basic WLabel setup
* WTrackProperty overrides `setup` and expects the widget to have a property node to connect to a valid property etc.

Reply:
I'm not sure, I'm not familiar enough with the code in question.
I don't think either `WTrackProperty` or `WTrackText` should inherit from the other. Instead the common functionality should be implemented in a base class and only the differences (eg the `setup` method) should live in `WTrackText` and `WTrackProperty`. Does that make sense? I haven't looked too close at the code so I can't judge how feasible this is.

Reply:
I understand. In that case the base class would handle track loading, dnd & context menu events, and it would be of no practical use (no text is displayed, much like `WTrackWidgetGroup`).
The derived classes would only need to define `setup`, `updateLabel` and `mouseDoubleClickEvent`.

Actually, with this PR, we could deprecate `WTrackText` (just migrate to WTrackProperty in the official skins) and the outcome would be the same, though I didn't consider that since in the first place because WTrackText is probably used in custom skins.

If I would have thought of that, I'd also had been obvious that we can drop WTrackText and use WTrackProperty with just a few changes in legacyskinloader :man_facepalming: 
See last commit.

Reply:
No, that would have been too easy.
`Text` widget is already designed for two scenarios:
* if there's a `Text` defined that string is displayed (basically the same as WLabel)
* if there's a `Group` defined it shows "track info" (artist &/ title / file name) (and ignores `Text`, if defined)

I think my proposal from above can be implemented in legacyskinparser.
I'll take a look. If that's too cumbersome I propose to merge this a first step.

Reply:
Sure... I don't understand all of what you're saying because I'm not familiar enough with the skin system so I can't really form a well informed opinion. 
This PR doesn't worsen the code in anyway, so sure we can merge now and do the rest later.

Reply:
Ha, done!

Test for skin regression without the skin migration commit https://github.com/mixxxdj/mixxx/pull/12004/commits/f29580c8d62456d6ca43e6813d2bbf26cf65da31

Reply:
Testing without https://github.com/mixxxdj/mixxx/commit/f29580c8d62456d6ca43e6813d2bbf26cf65da31:
* all knob labels and fixed deck labels should be okay
* the only place where Text was used from 'track info' are all preview decks and samplers Shade